feature/platform/transceiver/otg_test/zr_temperature_test/zr_temperature_test.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zr_temperature_test
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"github.com/openconfig/ygnmi/ygnmi"
+	"github.com/openconfig/ygot/ygot"
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// Topology:
+//   dut1:port1 <--> port1:dut2
+//
+
+func TestZRTemperatureState(t *testing.T) {
+	dut1 := ondatra.DUT(t, "dut1")
+	dp1 := dut1.Port(t, "port1")
+	t.Logf("dut1: %v", dut1)
+	t.Logf("dut1 dp1 name: %v", dp1.Name())
+	// Check if TRANSCEIVER is of type 400ZR
+	if dp1.PMD() != ondatra.PMD400GBASEZR {
+		t.Fatalf("%s Transceiver is not 400ZR its of type: %v", dp1, dp1.PMD())
+	}
+	component1 := gnmi.OC().Component(dp1.Name())
+	temperatureInstantdp1 := gnmi.Get(t, dut1, component1.Temperature().Instant().State())
```

Comment:
throughout the test, `gnmi.Get` is used (which does a sub once query), however, based on the description, I believe this should be a `SAMPLE` subscription instead, e.g.
https://github.com/openconfig/featureprofiles/blob/a889483110c5e6e7371b57c8313c4a2d2cbd7db6/feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go#L89

Reply:
Hi Sergey,

Updated the script as suggested please review and let me know any further changes required.

Thanks

Reply:
I was curious about this as well. The intention is to open a single subscription channel at the start of the test and monitor that the device streams changes over that channel as different events occur (i.e. config changes). Is that what happens? I wasn't sure if the test currently is initiating a new subscription channel every time it reads data, which essentially just mimics the previous Get() behavior. Thanks.

Reply:
Hi Eric, based on previous comment have removed the Get call and using sample subscription as suggested, please review the latest script